test/test_cosmology.py

```diff
@@ -33,5 +33,13 @@ def test_comoving_dist():
         assert np.allclose(vals, vals_ref, rtol, atol)
 
 
+def test_to_method_flatlambdacdm():
+    cosmo = CausticFlatLambdaCDM()
```

Comment:
I think everything defaults to float32, but perhaps you should explicitly construct it as float32, test that everything is float32, then call `to` and test that everything is float64

Reply:
This test was meant only to make sure private tensor of FlatLambdaCDM are sent to dtype and device. Please refer to `test_to_method` in the file `test_parametrized` for actual `.to()` method test. 

For this specific case, the private tensors are explictly constructed as `float32` tensors in the `__init__` method. See `FlatLambdaCDM` class.